src/components/project/project.service.ts

```diff
@@ -1200,4 +1243,25 @@ export class ProjectService {
       ).every((n) => n)
     );
   }
+
+  protected async validateOtherResourceId(
+    ids: string[],
+    label: string,
+    resourceField: string,
+    errMsg: string
+  ): Promise<boolean> {
+    for (const id of ids) {
+      const result = await this.db
+        .query()
+        .match([node('node', label, { id })])
+        .return('node')
+        .first();
+
+      if (!result) {
+        throw new NotFoundException(errMsg, `project.${resourceField}`);
```

Comment:
If the input is actually an array we should specify which index has failed. i.e. `project.otherLocationIds[3]`

```suggestion
        throw new NotFoundException(errMsg, `project.${resourceField}${Array.isArray(ids) ? `[${index}]`: ''}`);
```